src/compiler/checker.ts

```diff
@@ -17404,7 +17398,19 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function isPatternLiteralPlaceholderType(type: Type): boolean {
-        return !!(type.flags & (TypeFlags.Any | TypeFlags.String | TypeFlags.Number | TypeFlags.BigInt)) || isPatternLiteralType(type);
+        if (type.flags & TypeFlags.Intersection) {
+            return some((type as IntersectionType).types, isPatternLiteralPlaceholderType);

```

Comment:
Hm, we don't want `"a"` to be a valid placeholder type, but we do want `"a" & {brand}` to be, so I'd move the `Primitive`-allowing check into the intersection check. So
```suggestion
            return some((type as IntersectionType).types, t => !!(t.flags & (TypeFlags.Literal | TypeFlags.Null | TypeFlags.Undefined)) || isPatternLiteralPlaceholderType(t));
```
(and also use `TypeFlags.Literal | TypeFlags.Null | TypeFlags.Undefined` instead of `Primitive` to match what we use within the other branch of `getTemplateLiteralType`.

Reply:
Though the `Null` and `Undefined` may be unnecessary, since I think our construction rules mean you can't brand them like you can other literal types.